src/rpc/blockchain.cpp

```diff
@@ -744,7 +744,7 @@ static RPCHelpMan pruneblockchain()
             "                  to prune blocks whose block time is at least 2 hours older than the provided timestamp."},
                 },
                 RPCResult{
-                    RPCResult::Type::NUM, "", "Height of the last block pruned"},
+                    RPCResult::Type::NUM, "", "The height of the first (lowest) block that is still stored, after pruning"},
```

Comment:
This is actually a bug. From 0.14 (2017 Mar) until before 0.19 (2019 Nov), the height of the last block pruned was returned, subject to a bug if there were blocks left unpruned due to sharing files with later blocks.

In #15991, this was "fixed" to the current implementation, introducing the new bug you're documenting here.

Since the user provides the parameter as a block to include in pruning, I have a slight preference for fixing the behaviour to match the documentation. See #24629

Reply:
Ok, makes sense. Dropped the commit and will take a look at #24629.